converters/fromZigbee.js

```diff
@@ -2722,6 +2722,18 @@ const converters = {
             return result;
         },
     },
+    viessmann_thermostat_att_report: {
+        cluster: 'hvacThermostat',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result = converters.thermostat_att_report.convert(model, msg, publish, options, meta);
+            // ViessMann ViCare TRV seems to report pIHeatingDemand between 0 and 5 already
+            if (typeof msg.data['pIHeatingDemand'] == 'number') {
+                result.pi_heating_demand = precisionRound(msg.data['pIHeatingDemand'], 0);
```

Comment:
If I understand correctly, for most thermostats `pi_heating_demand` is a value between 0-100. For this thermostat it is a value between 0 - 5. Would it be possible to convert this to the 0-100 range? E.g. 0 = 0, 1 = 20, 2 = 40, 3 = 60, 4 = 80, 5 = 100?

Reply:
Not 100% sure, ... I think for now we should pass the value as is.

I have only observed 1 and 5 myself while testing, 1 seems to indicate closed and 5 indicate open. I have not yet observed values in between 1 and 5 (neither 0)

Perhaps:
0 = ???
1 = 0%
2 = 25%
3 = 50%
4 = 75%
5 = 100%

As when I was testing I was setting it to 8 and 30 respectively. setting it around 20 or 15 didn't seem to change the valve it either remained at 5 for a bit then jumped to 1. Or the other way around. So passing the value as-is until we understand it better might be the best thing. If we have more uses and we are reporting modified values, perhaps we can find a pattern.

Reply:
```
zigbee2mqtt:info  2020-06-11 22:44:53: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.77,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'    [45/628]
zigbee2mqtt:info  2020-06-11 22:57:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.77,"linkquality":42,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:00:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":42,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:02:49: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:12:10: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:15:10: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:27:19: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:30:19: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:42:27: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:44:55: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:45:27: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-11 23:57:36: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:00:36: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:03:24: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:12:45: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:15:45: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:27:54: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:30:54: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:43:02: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:46:02: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 00:58:11: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 01:01:11: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 01:04:00: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:13:20: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:16:20: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.82,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:28:28: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.82,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:31:28: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:43:37: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.81,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:46:37: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 01:58:46: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":55,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 02:01:46: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":74}'
zigbee2mqtt:info  2020-06-12 02:04:35: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:13:55: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:16:55: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":42,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:29:03: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:32:03: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:44:12: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:47:12: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 02:59:21: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.8,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:02:21: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:05:10: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:14:29: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:17:29: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:29:38: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:32:38: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:44:47: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.79,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:47:47: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.78,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 03:59:56: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.78,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 04:02:56: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.77,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":0,"battery":75}'
zigbee2mqtt:info  2020-06-12 04:05:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.77,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":1,"battery":75}'
zigbee2mqtt:info  2020-06-12 04:05:45: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.77,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":1,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:18:04: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.76,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":1,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:20:09: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.76,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":1,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:24:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.76,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:27:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.76,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":4,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:33:13: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.75,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":4,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:42:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.75,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:48:22: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.75,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":74}'
zigbee2mqtt:info  2020-06-12 04:57:09: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.75,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":74}'
zigbee2mqtt:info  2020-06-12 05:03:30: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.74,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":74}'
zigbee2mqtt:info  2020-06-12 05:06:20: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.74,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:12:18: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.74,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:18:39: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.71,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:27:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.46,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:27:27: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.46,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:36:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":19.74,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:38:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":20,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:40:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":20.3,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:42:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":20.62,"linkquality":42,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:42:36: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":20.62,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:44:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":20.93,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:46:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.21,"linkquality":47,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:49:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.58,"linkquality":36,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:52:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.88,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":5,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:53:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.88,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":4,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:54:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.88,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 05:56:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":22.13,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:06:55: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":22.13,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:09:09: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":22.13,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:11:09: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.92,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:19:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.67,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:24:18: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.67,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:29:01: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.41,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:39:27: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.41,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:44:09: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.22,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:54:35: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.22,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 06:59:18: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.08,"linkquality":34,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 07:07:29: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.08,"linkquality":39,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'
zigbee2mqtt:info  2020-06-12 07:09:44: MQTT publish: topic 'zigbee2mqtt/bedroom/radiator', payload '{"local_temperature":21.08,"linkquality":57,"occupied_heating_setpoint":16.5,"pi_heating_demand":2,"battery":75}'```

Some stuff from last night, I don't see any sane pattern in it at this point. (This is just the raw value)

Reply:
If we know that 1 = 0 and 5 = 100, why not just return this? (and don't return anything for other values). This seems to make more sense then returning 1 or 5.

Reply:
It's just a guess though, and the values don't make sense at all, it was at 2 for a lot fo the night while it was warmer than the setpoint? And Idid not hear the motor, so i don't think it was on.

Reply:
If we don't know what the value means, I think we should leave it out entirely, it makes no sense to provide users with strange value, this will only cause confusion.

Reply:
My idea was to remove the broken step and report it as-is for now, so we can figure it out eventually. Removing the reporting configuration also stops it from sending it at all.

Perhaps we can configure the reporting and just log the value to debug? Would that be acceptable?

Reply:
Yes that's acceptable

Reply:
Hmm seems like all existing ones use `console.log()` to log, but that would not be hidden on the default log level right?

Reply:
Going to try with 
```
diff --git a/converters/fromZigbee.js b/converters/fromZigbee.js
index 8060eae..0906241 100644
--- a/converters/fromZigbee.js
+++ b/converters/fromZigbee.js
@@ -2729,7 +2729,7 @@ const converters = {
             const result = converters.thermostat_att_report.convert(model, msg, publish, options, meta);
             // ViessMann ViCare TRV seems to report pIHeatingDemand between 0 and 5 already
             if (typeof msg.data['pIHeatingDemand'] == 'number') {
-                result.pi_heating_demand = precisionRound(msg.data['pIHeatingDemand'], 0);
+                meta.logger.debug(`viessmann: piHeatingDemand reported as ${msg.data['pIHeatingDemand']}`);
             }
             return result;
         },
```

🤞

Reply:
That didn't work, but the good thing is... I just need to remove the value from the result as debug will already log it in the raw att_report... let me clean up and push

Reply:
Although it would be nice to have logger available in meta in the future though.

```
zigbee2mqtt:error 2020-06-12 21:06:24: Failed to call 'Receive' 'onZigbeeEvent' (TypeError: Cannot read property 'debug' of undefined
    at Object.convert (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/converters/fromZigbee.js:2734:29)
    at converters.forEach (/opt/zigbee2mqtt/lib/extension/receive.js:168:41)
    at Array.forEach (<anonymous>)
    at Receive.onZigbeeEvent (/opt/zigbee2mqtt/lib/extension/receive.js:166:20)
    at Controller.callExtensionMethod (/opt/zigbee2mqtt/lib/controller.js:352:44)
    at Controller.onZigbeeEvent (/opt/zigbee2mqtt/lib/controller.js:238:14)
    at Zigbee.emit (events.js:182:13)
    at Controller.herdsman.on (/opt/zigbee2mqtt/lib/zigbee.js:67:52)
    at Controller.emit (events.js:182:13)
    at Controller.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/controller.js:496:22))
```

Reply:
Added in latest dev branch (`meta.logger`)